sdk/src/client/secret/types.rs

```diff
@@ -195,7 +196,13 @@ impl From<&InputSigningData> for InputSigningDataDto {
         Self {
             output: OutputDto::from(&input.output),
             output_metadata: OutputMetadataDto::from(&input.output_metadata),
-            chain: input.chain.clone(),
+            chain: input.chain.as_ref().map(|chain| {
+                chain
+                    .segments()
+                    .iter()
+                    .map(|seg| u32::from_be_bytes(seg.bs()) ^ Segment::HARDEN_MASK)
```

Comment:
Is this `u32::from_be_bytes(seg.bs())` still necessary?

Reply:
Can you put a small TODO ?

Reply:
Actually it's private, so without exposing it in crypto.rs it's not possible

Reply:
What is the purpose of ` ^ Segment::HARDEN_MASK`? It'll convert non-hardened segment value to a hardened one and vice versa. Is that the intent? If you need to ensure that the segment is hardened then use `|  Segment::HARDEN_MASK`. If you need to ensure that the segment is non-hardened then use `& !Segment::HARDEN_MASK`.

Reply:
Thanks, used `& !Segment::HARDEN_MASK`
I think here we always want the non hardened version because that's the values the users usually provide (since it needs to be hardened when using it with ed25519, it doesn't really matter)